### DISCUSSION TITLE
Change meta of current page imperatively

### DISCUSSION THREAD

**Discussion by kaguya3222** [nuxt/nuxt#9890] [Questions (Nuxt 2)]
Hello! 

I want  `og:description`  meta tag to be equal `description`  meta tag on every page.

Something like this:
```
<meta name="description" content="custom page description">
<meta name="og:description" content="custom page description">
```

Is there any way to achieve this without writing `og:description` on every page? 

I don't want to write something like this on every page:
```
head() {
  return {
    meta:  [
       {
         name: 'description',
         hid: 'description',
         content: 'custom page description'
       },
       {
         name: 'og:description',
         hid: 'og:description',
         content: 'custom page description'
       }
    ]
  }
}
```

I want to write only `description` tag for the current page, and it would be cool if I could write a plugin that implements the logic of auto replacing the content of `og:description` tag to the content of `description` tag, but I can't realize how to get meta of the current page in `middleware`/`plugin`/`module`.

I tried to create a plugin and do something like this, but this is not working:
```
export default function ({ app }) {
  const currentDescription = app.head.meta.find(tag => tag.hid === 'description') // returns description that is set in nuxt.config.js head, not on current page
  app.head.meta.push({ name: 'og:description, hid: 'description', content: currentDescription.content  })
}
```

**Comment by kaguya3222** [comment]
up

**Comment by ghost** [comment]
Hi, use this method:
https://vuedose.tips/optimize-seo-and-social-media-sharing-in-a-nuxt-blog

**Reply by kaguya3222** [reply]
Hello, thanks for your answer, but your link is 404

**Comment by ManasMadrecha** [comment]
- You can create an SEO component and add it in all your pages and send props to it, e.g., description, title, image, etc.. 
- This component's template will be a blank `<div></div>`. But, this component's script will have a detailed `head` info, using the props being sent to it from the pages. 
- Then, in the pages that uses this component props, don't add `head` info. Just send the props once. 
- This way, you will only need to create this component once, and you can add SEO info easily.
